### PR TITLE
Client side record permissions

### PR DESCRIPTION
**Pull request by sgiaccio**

### CODE REVIEW

File: webapp/appModules/data/records/components/recordView.js [hunk 1]
```diff
@@ -45,19 +44,22 @@ class RecordView extends React.Component {
   }
 
   render () {
-    const {recordLoaded, preview} = this.props
+    const {recordLoaded, preview, user, record} = this.props
 
     return recordLoaded
-      ? <SurveyFormView draft={preview} preview={preview} edit={false} entry={true}/>
+      ? <SurveyFormView draft={preview} preview={preview} edit={false} entry={true} canEditRecord={canEditRecord(user, record)}/>
```
Comment (fefrain) [inline webapp/appModules/data/records/components/recordView.js]: do not pass user and record to props, calculate "canEditRecord" prop in mapStateToProps

File: webapp/appModules/data/records/components/recordsView.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ import TablePaginator from '../../../../commonComponents/table/tablePaginator'
 
 import Survey from '../../../../../common/survey/survey'
 import NodeDef from '../../../../../common/survey/nodeDef'
+import { canEditRecord } from '../../../../../common/auth/authManager'
```
Comment (fefrain) [inline webapp/appModules/data/records/components/recordsView.js]: our convention is to import the entire object and call function from it. e.g. AuthManager.canEditRecord

File: webapp/appModules/data/records/components/recordsView.js [hunk 3]
```diff
@@ -34,14 +36,20 @@ const RecordRow = ({idx, offset, record, style, nodeDefKeys}) => (
     <div>{getRelativeDate(record.dateModified)}</div>
     <div>{record.ownerName}</div>
     <div>
-      <Link to={appModuleUri(dataModules.record) + record.uuid} className="btn btn-s btn-of-light-xs">
-        <span className="icon icon-pencil2 icon-12px"></span>
-      </Link>
+      {
+        canEdit
```
Comment (fefrain) [inline webapp/appModules/data/records/components/recordsView.js]: if the change is only in the icon, make only the icon class dynamic

File: webapp/appModules/surveyForm/nodeDefs/components/nodeDefTableBody.js [hunk 4]
```diff
@@ -83,7 +83,11 @@ class NodeDefMultipleTableBody extends React.Component {
           <span className="values-summary">{this.state.nodeValues}</span>
           <button className="btn-s btn-of-light-xs"
                   onClick={() => this.toggleEditDialogOpen(true)}>
-            <span className="icon icon-pencil2 icon-12px"/>
+            {
+              this.props.canEditRecord
+                ? <span className="icon icon-pencil2 icon-12px" />
```
Comment (fefrain) [inline webapp/appModules/surveyForm/nodeDefs/components/nodeDefTableBody.js]: same as above, make only class name dynamic

File: webapp/appModules/data/records/components/recordView.js (hunk 1 above)
Comment (minotogna) [reply webapp/appModules/data/records/components/recordView.js]: user and   record are still in mapStateToProps, do not pass them to props,

**Pull request merged**